### PR TITLE
Working buttons in Shop Screen

### PR DESCRIPTION
**Pull request by btm505**
Added ability to press the buttons within the shop screen.

- [TEMP] Button to give player free money
- Button to buy ammo
- Button to continue to powerup screen

Added new Powerup Screen where player can purchase power ups (actual power ups not yet implemented).

Added ability to press ESC key to return through screen pages.

image
image

### CODE REVIEW

File: core/src/com/mygdx/game/Components/Pirate.java [hunk 1]
```diff
@@ -80,10 +80,10 @@ public void shoot(Vector2 dir) {
     /**
      * Adds ammo
      *
-     * @param ammo amount to add
+     * @param newAmmo amount to add
```
Comment (printincident) [inline core/src/com/mygdx/game/Components/Pirate.java:83]: Why is the variable Ammo changed?

File: core/src/com/mygdx/game/Components/Pirate.java [hunk 2]
```diff
@@ -80,10 +80,10 @@ public void shoot(Vector2 dir) {
...
      */
-    public void reload(int ammo) {
-        this.ammo += ammo;
+    public static void addAmmo(int newAmmo) {
```
Comment (printincident) [inline core/src/com/mygdx/game/Components/Pirate.java:85]: Probably a good idea changing this but would it be better to simply have a setter?

**Review by printincident — APPROVED** [review]
Looks good see comments

**Review by dom-hall — CHANGES_REQUESTED** [review]
Can we document the changes made to their code in issues? Like private -> public with some justification etc.

File: core/src/com/mygdx/game/Components/Pirate.java (hunk 1 above)
Comment (btm505) [reply core/src/com/mygdx/game/Components/Pirate.java:83]: There was a conflict with using the same variable name 'ammo' for the new ammo to be added to the existing ammo, as changing the function to static meant not being able to use the "this" reference.

Issue simply resolved by not naming them the same thing.

File: core/src/com/mygdx/game/Components/Pirate.java (hunk 2 above)
Comment (btm505) [reply core/src/com/mygdx/game/Components/Pirate.java:85]: Yeah I thought that too; only renamed their existing function here so it matched with similar functions.

(e.g. 'addPlunder' and 'addHealth')

**Review by dom-hall — APPROVED** [review]

**Pull request merged**